discord_quote/discord_quote/discord_quote.py

```diff
@@ -517,6 +609,237 @@ def pred(m):
+                        'pin',
+                        ctx.message.channel.name,
+                        msg_target,
+                        msg_id]))
+
+    try:
+        # Retrieve the message
+        msg_ = await ctx.guild.get_channel(channel_id).fetch_message(msg_id)
+        log.info(log_msg(['retrieved_quote',
+                      msg_id,
+                      ctx.guild.get_channel(channel_id).name,
+                      msg_.author.name,
+                      msg_.created_at.strftime("%Y-%m-%d %H:%M:%S"),
+                      ctx.message.author.name,
+                      msg_.clean_content]))
+
+        # Store the message
+        row = [alias, msg_.jump_url, ctx.message.author.name, ctx.message.created_at]
+        db_execute(
+                f"""INSERT INTO pins VALUES (
+                "{alias}",
+                "{msg_.jump_url}",
+                "{ctx.message.author.name}",
+                "{ctx.message.created_at}"
+            )
+            """)
+        
+        # Backup the database to S3
+        if bucket:
+            db_backup()
+
+        # Get, or create a webhook for the context channel
+        hook = await _get_hook(ctx, ctx.channel.id)
+
+        # Use WebHooks if possible
+        if hook:
+            payload = (
+                f"**{ctx.message.author.name}** just pinned " +
+                f"**{msg_.author.name}'s** [message](<{msg_.jump_url}>) to "+
+                f"*{alias}*"
+            )
+
+            out = await hook.send(
+                content=payload,
+                username=ctx.guild.me.name,
+                avatar_url=str(ctx.guild.me.avatar_url),
+                wait=True
+            )
+
+            log.info(log_msg(['sent_webhook_message',
+                              'pin successful',
+                              ctx.message.channel.name]))
+
+        else:
+            payload = (
+                f"**{ctx.message.author.name}** just pinned " +
+                f"**{msg_.author.name}'s** message to "+
+                f"*{alias}*"
+            )
+            await ctx.channel.send(payload)
+            log.info(log_msg(['sent_message',
+                              'pin successful',
+                              ctx.message.channel.name]))
+
+    except discord.errors.HTTPException as e:
+        log.warning(['msg_not_found', msg_id, ctx.message.author.mention, e])
+
+        # Return error if message not found.
+        await ctx.channel.send(
+            f"Couldn't find ({msg_id}) from channel {channel_id}. " +
+            f"Requested by {ctx.message.author.name}."
+        )
+
+        log.info(log_msg(['sent_message',
+                          'invalid_pin_request',
+                          ctx.message.channel.name]))
+
+@bot.command(aliases=['g'])
+async def get(ctx, *, request:str):
+    pin = db_execute(
+            f"SELECT msg_url FROM pins WHERE alias=\"{request.lower()}\""
+    )
+    if len(pin) > 0:
```

Comment:
this probably could be `==1` as extra protection against doubles?

Reply:
I want this to still work even when there are duplicates. Rare case where I want a silent failure.